group/edwards25519/scalar.go

```diff
@@ -175,8 +181,8 @@ func (s *scalar) MarshalBinary() ([]byte, error) {
 
 // UnmarshalBinary reads the binary representation of a scalar.
 func (s *scalar) UnmarshalBinary(buf []byte) error {
-	if len(buf) != 32 {
-		return errors.New("wrong size buffer")
+	if len(buf) < 32 {
```

Comment:
Why is that ? UnmarshalBinary should only consider valid scalars, i.e. 32 bytes. SetBytes takes whatever and reduces it to 32 bytes.

Reply:
I think this is left over from when I was trying to send the results of the hash (64 bytes) straight into UnmarshalBinary. Will revert.